### DISCUSSION TITLE
add bulb profiles to Arch

### DISCUSSION THREAD

**Discussion by raoul29140** [FreeCAD/FreeCAD#14899] [Code contribution discussions]
Hello all,
I'm a french total newbie on GitHub.
I've wrote some code to add Bulb Profiles (naval shapes) to Arch module but have no idea how to push this code to the next release to be integrated.
![BulbFlat_Profile](https://github.com/FreeCAD/FreeCAD/assets/173637828/fe11885e-7618-48f7-8e8e-d13635d38217)

2 files are modified :
-ArchProfile.py : added "BF" section with code to draw the section
-profiles.csv : added "BulbFlat" section with european size datas
please find attached files for thoses who want to use them.
[profiles.csv](https://github.com/user-attachments/files/15946381/profiles.csv)
[ArchProfile.py.gz](https://github.com/user-attachments/files/15946398/ArchProfile.py.gz)
files are to put in folder
~usr/Mod/Arch for ArchProfile.py
~usr/share/Mod/Arch/Presets for profiles.csv

**Comment by pierreporte** [comment]
We’re currently in feature freeze so I doubt it can be integrated. Since this is arch workbench (now called BIM), @yorikvanhavre may have a better answer and could help you make a PR.

**Comment by yorikvanhavre** [comment]
Yes, this will have to wait until after 1.0 release. We can add your changes manually, but I would recommend you do it yourself, because then your name will be part of the FreeCAD code history forever, and you will be an official FreeCAD developer ;)

It's not hard, if you have never worked with Git/GitHub before we can walk you through it. Basically it involves:

- Forking the FreeCAD repo (use the "fork" button on the top right corner)
- Cloning your fork on your own computer (on Linux one usually uses the terminal, on Windows/Mac there are many Git applications, Github itself has [one](https://desktop.github.com/).
- Creating a new branch (so your changes go in a separate branch, not in your "main" branch)
- Applying your changes (basically replacing the files above)
- Committing your changes
- Pushing your branch to your GitHub fork
- Creating a Pull Request from your fork/branch to the FreeCAD repo

**Comment by raoul29140** [comment]
hello @pierreporte and @yorikvanhavre,
thank you for your answer
what a greate perspective to be part of FreeCAD forever ! ;-)
ok, I will do all these steps (i work on linux debian)

**Comment by yorikvanhavre** [comment]
There are several Git GUI applications on linux (gitg, gitkraken,...) but 99% of people on linux use the git command in the terminal directly, it's often easier than these GUI apps. Shout if you get stuck and need help!

**Reply by raoul29140** [reply]
Yes, I use linux for several years, and I also prefere command line tools.
I use last stable "appimage" of freecad, more convenient to avoid all dependancies issue, so I unpacked the appimage  to modify the files.
I will try to compile current release with my modified files to check if it works.
